### ISSUE TITLE
How to map train_edge.py files to .step source data？

### ISSUE THREAD

**Issue by zctlovesy0313** [AndrewColligan/CADNet#2]
How do I map the result labels predicted by the model to the STEP model for visualization?
<img width="890" alt="c59b01277c5a5f42243c6d6f55d540a" src="https://github.com/user-attachments/assets/865c9dab-393f-41e7-abd9-556e6583e8dd">
The shape of logits is 704, what does 704 represent?
The events I run are:https://www.sciencedirect.com/science/article/pii/S0010448522000240#appB

**Comment by AndrewColligan** [comment]
Hi, in the Hierarchical CADNet repo there is a script called [test_and_save.py](https://gitlab.com/qub_femg/machine-learning/hierarchical-cadnet/-/blob/master/test_and_save.py?ref_type=heads) that can be used to predict the machining features for individual CAD models and save the results to a CAD model (with a label per B-Rep face) using a trained NN. 

The logit shape is the total number of B-Rep faces predicted per minibatch. The minibatch is made up of a number of the graphs of different CAD models, each of these CAD models will have a different number of B-Rep faces associated with them. Therefore, logit shape will be different each minibatch as the number of B-Rep faces of each CAD model will differ.

> Logit shape = total number of B-Rep faces in each CAD model in the minibatch

Hope this helps 😄

**Comment by zctlovesy0313** [comment]
Thank you! Another question: how to get string data representing different polygon labels, e.g. [1,24,2,12,6,......]. Each number represents a different face.And can I get, for example, the relative position of a hole relative to the entire drawing, so that I can generate the machining path of the CNC machine? Do you have any good suggestions, please?
